k9mail/src/main/java/com/fsck/k9/view/MessageCryptoDisplayStatus.java

```diff
@@ -237,12 +252,15 @@ private static MessageCryptoDisplayStatus getDisplayStatusForPgpResult(CryptoRes
             }
         }
 
+        final long delta = signatureResult.getSignatureTimestamp().getTime() - messageSentDate.getTime();
```

Comment:
The mechanism would be similar to the `SENDER_ADDRESS` one, returning its own status for the additional given context or not. So this wouldn't affect other clients :)

Also the issue indeed exists in xmpp, at least if you're talking about [XEP-0373](http://xmpp.org/extensions/xep-0373.html), since there is a timestamp in the <signcrypt> tag that may or may not match the one in the signature itself. Or am I missing something there?